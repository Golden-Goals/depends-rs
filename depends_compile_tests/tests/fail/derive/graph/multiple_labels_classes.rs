use depends::derives::Graph;
#[allow(unused_imports)]
use examples::*;

// Multiple labels for an edge.
#[derive(Graph)]
#[depends(
    digraph Dag {
        comments [label="Comments"];
        posts [label="Posts"];

        comments_posts [label="CommentsPosts"];

        comments -> comments_posts [label="TrackCommentPostIds", class="CommentTracker"];
        posts -> comments_posts [label="AnotherLabel", class="CommentTracker"];
    }
)]
struct Graph1 {}

// Multiple classes for an edge.
#[derive(Graph)]
#[depends(
    digraph Dag {
        comments [label="Comments"];
        posts [label="Posts"];

        comments_posts [label="CommentsPosts"];

        comments -> comments_posts [label="TrackCommentPostIds", class="CommentTracker"];
        posts -> comments_posts [label="TrackCommentPostIds", class="AnotherClass"];
    }
)]
struct Graph2 {}

fn main() {}
