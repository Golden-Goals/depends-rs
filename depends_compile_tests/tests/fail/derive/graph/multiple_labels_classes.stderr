error: Multiple values for `label` and `class` on edges to node: {("AnotherLabel", Some("CommentTracker")), ("TrackCommentPostIds", Some("CommentTracker"))}.
  --> tests/fail/derive/graph/multiple_labels_classes.rs:12:9
   |
12 |         comments_posts [label="CommentsPosts"];
   |         ^^^^^^^^^^^^^^

error: Multiple values for `label` and `class` on edges to node: {("TrackCommentPostIds", Some("AnotherClass")), ("TrackCommentPostIds", Some("CommentTracker"))}.
  --> tests/fail/derive/graph/multiple_labels_classes.rs:27:9
   |
27 |         comments_posts [label="CommentsPosts"];
   |         ^^^^^^^^^^^^^^
